doc/sphinx/source/howto/tune-for-deployment.rst

```diff
@@ -0,0 +1,171 @@
+be explored. MKL-DNN is automatically enabled as part of an nGraph build; you do 
+*not* need to add MKL-DNN separately on the framework side to use these 
+configuration settings.  
+
+* ``KMP_BLOCKTIME`` Sets the time, in milliseconds, that a thread should wait 
+  after completing the execution of a parallel region, before sleeping.
+* ``KMP_AFFINITY`` Enables the runtime library to bind threads to physical 
+  processing units. 
+* ``KMP_SETTINGS`` Enables (``true``) or disables (``false``) the printing of 
+  OpenMP* runtime library environment variables during program execution.
+* ``OMP_NUM_THREADS`` Specifies the number of threads to use.
+
+
+Code deployment to nGraph-enabled Intel® Xeon®
+==============================================
+
+The list below includes recommendations on :term:`data layouts`, parameters, and 
+application configuration to achieve best performance running DNN workloads on 
+Intel® Xeon® (CPU processor) systems.
+
+Threading 
+---------
+
+The number of threads set by ``OMP_NUM_THREADS`` ought not exceed the number of 
+physical cores. The threads should be pinned to their respective physical cores 
+and actived as follows:
+
+* When ``HT=off``, ``KMP_AFFINITY=compact,granularity=fine``
+
+* When ``HT=on``, ``KMP_AFFINITY=compact,1,0,granularity=fine``
+
+
+Memory allocation 
+-----------------
+
+Buffer pointers should be aligned at the 64-byte boundary. NUMA policy should be 
+configured for local memory allocation (``numactl --localloc``)
+
+Convolution shapes
+^^^^^^^^^^^^^^^^^^
+* When **running inference, or training for forward-propagation and weight 
+  updates**,
+  
+  - the number of input channels should be 1, 3, or a multiple of SIMD-width (8 
+    for AVX2 systems, 16 for AVX512 systems). 
+  - the number of output channels should be a multiple of SIMD-width (8 for AVX2 
+    systems, 16 for AVX512 systems).
+
+* When **training backward propagation**, the number of input and output 
+  channels should be a multiple of SIMD-width (8 for AVX2 systems, 16 for AVX512 
+  systems),
+  
+  - padding should not exceed :math:`0.5x` where :math:`x` is the kernel size.
+  - kernel width should be less than 14.
+
+
+``OMP_NUM_THREADS``
+^^^^^^^^^^^^^^^^^^^
+
+The best resource for this configuration option is the `gnu.org site`_ 
+``OMP_NUM_THREADS`` defaults to the number of physical cores. To check the 
+number of cores on your system, you can run the following on the command-line to 
+see the details of your CPU: 
+
+.. code-block:: console
+
+   $ lscpu
+
+Adjusting this parameter beyond matching the number of cores can have an impact 
+when using Intel® Xeon Phi™ (Knights Landing) for some models. 
+
+Intra-op and inter-op parallelism 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* ``intra_op_parallelism_threads``
+* ``inter_op_parallelism_threads``
+
+Some frameworks attempt to use these settings to improve performance; however, 
```

Comment:
TensorFlow uses these settings to improve performance.

Reply:
Added